core/dbt/deps/tarball.py

```diff
@@ -42,23 +48,31 @@ def nice_version_name(self):
         return f"tarball (url: {self.tarball})"
 
     def _fetch_metadata(self, project, renderer):
-        """
-        recycle RegistryPackageMetadata so that we can use the install and
-        download_and_untar from RegistryPinnedPackage next.
-        build RegistryPackageMetadata from info passed via packages.yml since no
-        'metadata' service exists in this case.
-        """
-
-        dct = {
-            "name": self.package,
-            "packages": [],  # note: required by RegistryPackageMetadata
-            "downloads": {"tarball": self.tarball},
-        }
-
-        return RegistryPackageMetadata.from_dict(dct)
+        """Download and untar the project and parse metadata from the project folder."""
+        system.download(self.tarball, self.tar_path)
+        system.untar_package(self.tar_path,self.untarred_path)
```

Comment:
We should use the same `download_and_untar` + `connection_exception_retry` methods here. That way, if we need to add more things ([like `sha1` comparison for tarball contents](https://github.com/dbt-labs/dbt-core/issues/4637)), we can do it in one place.

IMO it also makes more sense for this method to include the check of the downloads directory path, to see whether the package has already been downloaded there. (The [default behavior](https://github.com/dbt-labs/dbt-core/blob/e24f9b3da722328e24ee10a06fa2e8d2ca73b3e6/core/dbt/deps/base.py#L31-L35) is to clear the downloads directory after every `dbt deps` run.)